client/src/partials/journal/modals/search.modal.html

```diff
@@ -87,6 +87,22 @@
         <div ng-messages-include="partials/templates/messages.tmpl.html"></div>
       </div>
     </div>
+
+    <div class="form-group" ng-class="{ 'has-error' : ModalForm.$submitted && ModalForm.trans_id.$invalid }">
+      <label class="control-label">
+        {{ 'FORM.LABELS.TRANSACTION' | translate }}
+      </label>
+      <span style="display:inline-block;" class="pull-right">
+        <a href ng-click="ModalCtrl.clear('display_name')" tabindex="-1">
```

Comment:
This clear button will not work properly because you are referencing `display_name` instead of `trans_id`.  It _must_ reference the model name.

You should also just use `bhClear` which recently landed, like this:
 ```html
<bh-clear on-clear="ModalCtrl.clear('trans_id')></bh-clear>
```
and remove all this extra code.